OpenRA.Game/Server/Server.cs

```diff
@@ -20,106 +20,65 @@
 using OpenRA.Primitives;
 using OpenRA.Support;
 
-using XTimer = System.Timers.Timer;
-
 namespace OpenRA.Server
 {
-	public enum ServerState : int
+	public enum ServerState
 	{
-		WaitingPlayers = 1,
-		GameStarted = 2,
-		ShuttingDown = 3
+		WaitingPlayers,
+		GameStarted,
+		ShuttingDown
 	}
 
 	public class Server
 	{
+		#region Members
+		
+		public readonly ModData ModData;
+		public readonly ServerSettings Settings;
+		public readonly Session LobbyInfo;
 		public readonly IPAddress Ip;
 		public readonly int Port;
-
-		int randomSeed;
 		public readonly MersenneTwister Random = new MersenneTwister();
+		public readonly List<string> TempBans = new List<string>();
 
 		// Valid player connections
-		public List<Connection> Conns = new List<Connection>();
+		readonly List<Connection> conns = new List<Connection>();
+		/*
+		 * Pre-validated player connections,
+		 * These are players that have connected to the server
+		 * but haven't sent a handshake request, so they're not registered
+		 * to play. This refers to "preConns" by the way, the variable declared below.
+		 * But not "conns", that's declared above, which refer to connections which
+		 * ARE valid. This is very confusing to some, it may take weeks if you
+		 * don't have a college degree.
+		*/
+		readonly List<Connection> preConns = new List<Connection>();
 
-		// Pre-verified player connections
-		public List<Connection> PreConns = new List<Connection>();
+		readonly TypeDictionary serverTraits = new TypeDictionary();
+		readonly TcpListener listener;
 
-		TcpListener listener = null;
+		public volatile ServerState State = new ServerState();
 
-		TypeDictionary serverTraits = new TypeDictionary();
-		public Session LobbyInfo;
+		public Map Map { get; set; }
 
-		public ServerSettings Settings;
-		public ModData ModData;
-		public Map Map;
-		XTimer gameTimeout;
+		public IEnumerable<Connection> Connections { get { return conns.ToList(); } }
```

Comment:
The `.ToList()` is needed because you can't add/remove items to a collection you're enumerating.  I think that it would be clearer to keep these inline instead of pulling this out in the name of de-duplication (and then suffering through arguments like the one I just had to read).